### PR TITLE
DIG-1555 & DIG-1595: Update deployment documentation

### PR DESCRIPTION
**Pull request by mshadbolt**
* Delete docs that are no longer relevant
* Added a new doc on how to back up data in CanDIG
* Added new doc about interacting with stack with Makefile
* Added new doc about production deployments
* Edits to install-candig doc

### CODE REVIEW

File: docs/backing-up-and-restoring-candig.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+1. Then restart the katsu and htsget services
+
+```bash
+docker start candigv2_katsu_1
+docker start candigv2_htsget_1
+```
+
+You should be able to see the restored data in the data portal.
+
+## Backing up Authorization data
```
Comment (mshadbolt) [inline docs/backing-up-and-restoring-candig.md]: Can @daisieh add how to back up auth data out of vault please?

**Comment by daisieh** [comment]
We should mention that production instances may want to keep any existing logs from tmp/logging in a safe place before wiping the system...

File: docs/interact-with-the-stack.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+> [!NOTE]
+> If there are changes that have changed the structure of the database or impacted the versions of other CANDIG_DATA_MODULES this way of rebuilding cannot be used.
+
+## Destructive Cleanup 
+
+Use the following steps to clean up running CanDIGv2 services in a docker-compose configuration. 
+
+> [!CAUTION] 
+> Note that these steps are destructive and will remove **ALL** logs, containers, secrets, volumes, networks, certs, and images. If you are using docker in a shared environment (i.e. with other non-CanDIGv2 containers running) please consider running the cleanup steps manually instead.
```
Comment (OrdiNeu) [inline docs/interact-with-the-stack.md:97]: Er -- the only one of these that can affect things outside of CanDIG is the clean-images step; all of the rest of the steps are either cleaning up something in the temp directory (logs, secrets), or specifically target `label=candigv2` (containers),  `dangling=true` (volumes), or refer to the `.env` (compose)

File: docs/interact-with-the-stack.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+## Rebuild entire stack from scratch
+
+1. Perform any backups of data necessary if in a non-testing environment. (see [backup and restore doc](backup-restore-candig.md) for detailed instructions.)
+
+2. Clean up the current containers with `make clean-all`
+
+3. When complete, build all containers again with `make build-all`
+
+
+## Troubleshooting
```
Comment (OrdiNeu) [inline docs/interact-with-the-stack.md:132]: Is it possible to link to our debugging page (and make it public) here? https://candig.atlassian.net/wiki/spaces/CA/pages/684130305/Debugging

File: docs/production-candig.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+### HAProxy - UHN & BCGSC
+
+At UHN, the candig.uhnresearch.ca domain is under a proxy, so requests to a specific service go through the following stack:
+
+```
+[ CLIENT ] ---> | HTTPS | ---> [ UHN PROXY ] ---> | HTTP | ---> [ CANDIG_PROD / TYK ] ---> | HTTP | ---> [ CANDIG_DATA_SERVICES ]
+```
+
+Specifically, the UHN proxy forwards all candig.uhnresearch.ca and candigauth.uhnresearch.ca requests (all ports) to candig1:5080 (tyk) and candig1:8080 (keycloak) respectively, thereby acting as a firewall.  All CanDIGv2 microservices can only be accessed through Tyk.
```
Comment (OrdiNeu) [inline docs/production-candig.md]: It technically only forwards on https connections -- conenctions on port 80 redirect to 443, and it otherwise doesn't respond to most other ports.
```suggestion
Specifically, the UHN proxy forwards all candig.uhnresearch.ca and candigauth.uhnresearch.ca requests (port 443) to candig1:5080 (tyk) and candig1:8080 (keycloak) respectively, thereby acting as a firewall.  All CanDIGv2 microservices can only be accessed through Tyk.
```

**Review by OrdiNeu — APPROVED** [review]
Looks great! Only a few comments:

**Review by SonQBChau — APPROVED** [review]
LGTM

**Review by CourtneyGosselin — CHANGES_REQUESTED** [review]
I have some concerns regarding potential issues with the production build, which are currently only documented in the production Confluence doc. It might be a good idea to move these details to the debugging doc, especially since Francis mentioned adding that section. Key points to include would be:

- The issue of having two proxies causing URL looping
- The need for an updated proxy to work with Python 3.10, as older versions may lead to cipher-related issues

Additionally, in the production Confluence doc, we discussed the province codes, which are currently found in the install-candig documentation. It could be more helpful to move this information to the production document, alongside the details on how to set up a site logo.

File: docs/interact-with-the-stack.md [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+
+### docker volumes not remade
+
+If you get an error where after cleaning an individual service, when composing, it gets stuck at 
+
+```
+waiting for x service to start ...
+```
+
+Use CTRL + c to exit the process then try running `make docker-volumes` and then try composing again with `make compose-<name of service>`
```
Comment (daisieh) [inline docs/interact-with-the-stack.md:153]: I keep meaning to add a bit to make compose that adds back any known external volumes...

File: docs/production-candig.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
+# Guide to CanDIG production deployments
+
+Apart from the basic steps in the [CanDIGv2 Install Guide](install-candig.md) to get the candig stack up and running, there are additional settings and security recommendations that need to set up in a production level environment. We provide the following as general advice, but it is important for all CanDIG deployers to also consult with their institutional infrastructure security personnel to ensure that their deployment meets the necessary level of data security.
+
+## Reverse Proxy & Firewall
+
+It is essential to setup a reverse proxy and firewall so that only specific ports are open to the internet. The software used for this is up to the deployer and is considered outside of the CanDIG stack. 
+
+Basically, the only ports that should be available are to tyk (443) and keycloak (80).
```
Comment (daisieh) [inline docs/production-candig.md:13]: This might be me being very dumb, but isn't keycloak 8080? or do we redirect to that?

**Review by daisieh — APPROVED** [review]
LGTM outside of the comments others made.

File: docs/production-candig.md [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+
+```
+[ CLIENT ] ---> | HTTPS | ---> [ UHN PROXY ] ---> | HTTP | ---> [ CANDIG_PROD / TYK ] ---> | HTTP | ---> [ CANDIG_DATA_SERVICES ]
+```
+
+Specifically, the UHN proxy forwards all candig.uhnresearch.ca and candigauth.uhnresearch.ca requests (all ports) to candig1:5080 (tyk) and candig1:8080 (keycloak) respectively, thereby acting as a firewall.  All CanDIGv2 microservices can only be accessed through Tyk.
+
+### OpenStack security group & nginx - C3G
+
+ OpenStack security group that allows access to ports 80 and 443 acts as a Firewall.
```
Comment (DavidBrownlee) [inline docs/production-candig.md]: Minor, but I'd rephrase to:
An OpenStack security group is applied as a firewall that allows ingress traffic to ports 80 and 443 only.

**Comment by DavidBrownlee** [comment]
There may be some updates when I install the new stable release.  The re-organization looks great!

**Review by DavidBrownlee — APPROVED** [review]

File: docs/interact-with-the-stack.md (hunk 3 above)
Comment (mshadbolt) [reply docs/interact-with-the-stack.md:132]: Is this document up-to-date with the current stack?

**Comment by mshadbolt** [comment]
> I have some concerns regarding potential issues with the production build, which are currently only documented in the production Confluence doc. It might be a good idea to move these details to the debugging doc, especially since Francis mentioned adding that section. Key points to include would be:
> 
> * The issue of having two proxies causing URL looping
> * The need for an updated proxy to work with Python 3.10, as older versions may lead to cipher-related issues
> 
> Additionally, in the production Confluence doc, we discussed the province codes, which are currently found in the install-candig documentation. It could be more helpful to move this information to the production document, alongside the details on how to set up a site logo.

@CourtneyGosselin are you able to have a go at moving the information to where you think it is most useful? I am not really that familiar with the issues and what is most useful for people who are trying to debug.

**Comment by CourtneyGosselin** [comment]
@mshadbolt Yeah, I can move the bug information to the debugging document. As for the site logo config and province codes, those are in the develop branch for the install doc right now. I think those may be better in the production document

**Comment by mshadbolt** [comment]
ok yep, I can move the province code stuff to the production doc

**Review by CourtneyGosselin — APPROVED** [review]
LGTM

File: docs/production-candig.md [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+```bash
+export TOKEN=$(echo $CURL_OUTPUT | grep -Eo 'access_token":"[a-zA-Z0-9._\-]+' | cut -d '"' -f3)
+```
+
+```bash
+curl -X GET $CANDIG_URL'/ingest/site-role/admin/email/site_admin@test.ca' -H 'Authorization: Bearer '$TOKEN
+```
+
+
+## Connecting Keycloak to institutional LDAP
```
Comment (mshadbolt) [inline docs/production-candig.md:174]: Is there any generic information we could add here for how people can connect their LDAP to keycloak? @OrdiNeu @SonQBChau ?

File: docs/backup-restore-candig.md [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+1. Restart the katsu and htsget services
+
+```bash
+docker start candigv2_katsu_1
+docker start candigv2_htsget_1
+```
+
+You should be able to see the restored data in the data portal.
+
+## Backing up Authorization data
```
Comment (mshadbolt) [inline docs/backup-restore-candig.md]: Are we likely to figure out how to do this @daisieh or any kind of way to get this info, e.g. using endpoints to dump the auth data if we can't get it out of vault?

File: docs/production-candig.md (hunk 6 above)
Comment (mshadbolt) [reply docs/production-candig.md:13]: I have seen different things in different docs and I don't understand the difference. @OrdiNeu ?

File: docs/production-candig.md (hunk 7 above)
Comment (mshadbolt) [reply docs/production-candig.md]: Thanks for the feedback, I have reworded as suggested.

File: docs/interact-with-the-stack.md (hunk 2 above)
Comment (mshadbolt) [reply docs/interact-with-the-stack.md:97]: This is something that was already in the docs, are you able to suggest a change to it if it isn't correct?

File: docs/production-candig.md (hunk 6 above)
Comment (mshadbolt) [reply docs/production-candig.md:13]: Would be good to have confirmation whether or not this is correct

File: docs/backing-up-and-restoring-candig.md (hunk 1 above)
Comment (mshadbolt) [reply docs/backing-up-and-restoring-candig.md]: This is done in latest commit
